src/mqtt_client/mod.rs

```diff
@@ -525,39 +548,80 @@ where
     }
 }
 
+#[repr(u8)]
+pub enum ResponseCode {
+    Ok = 0,
+    Continue = 1,
+
+    Error = 0xFF,
+}
+
 /// The payload of the MQTT response message to a settings update request.
 #[derive(Serialize)]
-pub struct SettingsResponse {
+struct Response<const N: usize> {
     code: u8,
-    msg: String<64>,
+    msg: String<N>,
 }
 
-impl SettingsResponse {
+impl<const N: usize> Response<N> {
     pub fn ok() -> Self {
         Self {
             msg: String::from("OK"),
-            code: 0,
+            code: ResponseCode::Ok as u8,
         }
     }
 
-    pub fn error(msg: String<64>) -> Self {
-        Self { code: 255, msg }
+    pub fn custom(code: ResponseCode, message: &str) -> Self {
+        let mut msg = String::new();
+        match msg.push_str(message) {
+            Ok(_) => Self {
+                code: code as u8,
+                msg,
+            },
+            Err(_) => Self::error("Message too long"),
+        }
+    }
+
+    pub fn error(message: &str) -> Self {
+        let mut msg = String::new();
+        msg.push_str(message).ok();
```

Comment:
That's the intent. If we can't serialize the message (because the user specified some very small max topic size), there's nothing more we can necessarily do and we fall back on just providing the response code. I'll document this.

Reply:
But shouldn't we do it like L602?